server/routes/api/contests.js

```diff
@@ -0,0 +1,347 @@
+const Contender = require('../../models/contests/Contender');
+const Contest = require('../../models/contests/Contest')
+
+const User = require('../../models/User');
+
+var requireRole = require('../middleware/Token').requireRole;
+var verifyUser = require('../middleware/Token').verifyUser;
+
+function toObject(arr) {
+  var rv = {};
+  for (var i = 0; i < arr.length; ++i)
+    if (arr[i] !== undefined) rv[i] = arr[i];
+  return rv;
+}
+
+module.exports = (app) => {
+    app.post('/api/contests/handle', verifyUser, function (req, res) {
+      var email = req.body.email;
+      var platform = req.body.platform;
+      var handle = req.body.handle;
+
+      if (!email || !platform || !handle) {
+        return res.status(400).send({
+          success: false,
+          message: 'Error: Email of user, Platform and Handle fields required.'
+        });
+      }
+
+      User.find({
+        email: email,
+        isDeleted: false,
+      }, (err, users) => {
+        if (err) {
+          return res.status(500).send({
+            success: false,
+            message: "Error: Server error"
+          });
+        }
+        if (users.length != 1) {
+          return res.status(404).send({
+            success: false,
+            message: 'Error: User not found.'
+          });
+        }
+        var user = users[0];
+        newHandle = {"platform": platform, "handle": handle};
+        Contender.findOneAndUpdate({
+          user: user._id
+        }, { $push: { handles: newHandle }}, (err, updatedContender) =>
+      {
+        if (err) {
+          return res.status(500).send({
+            success: false,
+            message: "Error: Server update error"
+          });
+        }
+      });
+
+        return res.status(200).send({
+          success: true,
+          message: 'Handle Added succesfully',
+        });
+      })
+    }), // Works
+
+    app.post('/api/contests/updateContender', requireRole("admin"), function (req, res) {
+      var email = req.body.email;
+      var contestName = req.body.contestName;
+      var score = req.body.score;
+      var rank = req.body.rank;
+      var currentRank = req.body.currentRank;
```

Comment:
What is the difference between rank and currentRank?

Reply:
Rank is for the rank as part of history. It is the rank for that contest. currentRank is the contender's overall rank in Alcoding itself.